bpf/lib/nodeport.h

```diff
@@ -3051,17 +3068,6 @@ static __always_inline int nodeport_lb4(struct __ctx_buff *ctx,
 #endif
 #endif /* ENABLE_DSR */
 
-#ifndef ENABLE_MASQUERADE_IPV4
-		/* When BPF-Masquerading is off, we can skip the revSNAT path via
-		 * CILIUM_CALL_IPV4_NODEPORT_NAT_INGRESS if:
-		 * - the packet is ICMP, or
-		 * - the packet is DSR-eligible (and thus not reply traffic by
-		 *   a remote backend that would require revSNAT / revDNAT)
-		 */
-		if (!is_svc_proto || nodeport_uses_dsr4(&tuple))
-			return CTX_ACT_OK;
-#endif /* ENABLE_MASQUERADE_IPV4 */
-
```

Comment:
For my understanding - we could have preserved the ICMP optimization here, correct?

Reply:
Hm, good point. I can restore that one.

Reply:
Is your long-term expectation that the annotation-mode will be enabled by default? Just thinking if we could have preserved more of the existing behaviour if `ENABLE_DSR_BYUSER` is not set.

But not sure if it's worth it, and how much compatibility we need to offer. I'd expect established DSR connections to get disrupted if a user suddenly disables the annotation-support?